src/main/asciidoc/reference/vault-secret-engines.adoc

```diff
@@ -0,0 +1,252 @@
+[[vault.core.secret-engines]]
+= Supporting for Vault's Secret Engines
+
+Spring Vault ships with several extensions to support Vault's various secret engines.
+
+Specifically, Spring Vault ships with extensions for:
+
+* <<vault.core.backends.kv1>>
+* <<vault.core.backends.kv2>>
+* <<vault.core.backends.pki>>
+* <<vault.core.backends.token>>
+* Transform (Enterprise Feature)
+* <<vault.core.backends.transit>>
+* System Backend
+
+All other backends can be used through methods on `VaultTemplate` directly (`VaultTemplate.read(…)`, `VaultTemplate.write(…)`).
+
+[[vault.core.backends.kv1]]
+== Key-Value Version 1 ("unversioned secrets")
+
+The `kv` secrets engine is used to store arbitrary secrets within the configured physical storage for Vault.
+
+When running the `kv` secrets engine non-versioned only the most recently written value for a key will be preserved.
+The benefits of non-versioned kv is a reduced storage size for each key since no additional metadata or history is stored.
+Additionally, requests going to a backend configured this way will be more performant because there will be fewer storage calls and no locking for any given request.
+
+Spring Vault ships with a dedicated Key-Value API to encapsulate differences between the individual Key-Value API implementations.
+`VaultKeyValueOperations` follows the Vault CLI design.
+That's the primary command line tool for Vault providing commands such as `vault kv get`, `vault kv put` and so on.
+
+This API can be used with both Key-Value engine versions by specifying the version and mount path.
+The following example uses the Key-Value version 1:
+
+====
+[source,java,indent=0]
+----
+include::../{example-root}/KeyValueV1.java[tags=keyValueApi]
+----
+====
+
+`VaultKeyValueOperations` supports all Key-Value operations such as `put`, `get`, `delete`, `list`.
+
+Alternatively, the API can be used through `VaultTemplate` because of its direct mapping and simple use, as keys and responses map directly to input and output keys.
+The following example illustrates writing and reading a secret at `mykey`.
+The `kv` secrets engine is mounted at `secret`:
+
+====
+[source,java,indent=0]
+----
+include::../{example-root}/KeyValueV1.java[tags=vaultOperations]
+----
+====
+
+You can find more details about the https://www.vaultproject.io/api-docs/secret/kv/kv-v1[Vault Key-Value version 1 API] in the Vault reference documentation.
+
+[[vault.core.backends.kv2]]
+== Key-Value Version 2 ("versioned secrets")
+
+The `kv` secrets engine can be run in one of two versions.
+This section explains using version 2. When running v2 of the `kv` backend a key can retain a configurable number of versions.
+The older versions' metadata and data can be retrieved.
+Additionally, Check-and-Set operations can be used to avoid overwriting data unintentionally.
```

Comment:
Change to "Additionally, you can use check-and-set operations to avoid unintentionally overwriting data."